src/lib.rs

```diff
@@ -0,0 +1,92 @@
+mod cli;
+pub mod config;
+mod types;
+
+use crate::config::Config;
+use actix_web::{
+    error,
+    web::{self, Json},
+    App, Error, HttpServer,
+};
+use std::{io::Write, path::PathBuf, process::Command};
+use types::{SolToUmlRequest, SolToUmlResponse};
+use uuid::Uuid;
+
+async fn sol_to_uml(
+    tmp_dir: web::Data<PathBuf>,
+    data: Json<SolToUmlRequest>,
+) -> Result<Json<SolToUmlResponse>, Error> {
+    let data = data.into_inner();
+
+    let unique_name = Uuid::new_v4().to_string();
+    let contract_dir = tmp_dir.get_ref().join(&unique_name);
+    let contract_dir =
+        web::block(move || std::fs::create_dir(contract_dir.clone()).map(|_| contract_dir))
+            .await
+            .map_err(error::ErrorInternalServerError)??;
+
+    for (name, content) in data.sources.iter() {
```

Comment:
```suggestion
    for (name, content) in data.sources {
```

It may iterate automatically (but do it by value in that case)